gcloud/bigquery/dataset.py

```diff
@@ -204,6 +273,28 @@ def _require_client(self, client):
             client = self._client
         return client
 
+    def _parse_access_grants(self, access):
+        """Parse a resource fragment into a schema field.
+
+        :type access: list of mappings
+        :param access: each mapping represents a single access grant
+
+        :rtype: list of :class:`AccessGrant`
+        :returns: a list of parsed grants
+        """
+        result = []
+        for grant in access:
+            role = grant['role']
+            if 'specialGroup' in grant:
+                entity_type = 'specialGroup'
+            elif 'groupByEmail' in grant:
+                entity_type = 'groupByEmail'
+            else:
```

Comment:
47199cb removes the hard-wired assumption that the entity types will match only the documented values.